src/client/Microsoft.Identity.Client/Platforms/Features/WebView2WebUi/WinFormsPanelWithWebView2.cs

```diff
@@ -65,23 +68,47 @@ internal class WinFormsPanelWithWebView2 : Form
             };
         }
 
-        public AuthorizationResult DisplayDialogAndInterceptUri()
+        public AuthorizationResult DisplayDialogAndInterceptUri(CancellationToken cancellationToken)
         {
             _webView2.CoreWebView2InitializationCompleted += WebView2Control_CoreWebView2InitializationCompleted;
             _webView2.NavigationStarting += WebView2Control_NavigationStarting;
 
             // Starts the navigation
             _webView2.Source = _startUri;
-            DisplayDialog();
+            DisplayDialog(cancellationToken);
 
             return _result;
         }
 
-        private void DisplayDialog()
+        private void DisplayDialog(CancellationToken cancellationToken)
         {
             DialogResult uiResult = DialogResult.None;
-            InvokeHandlingOwnerWindow(() => uiResult = ShowDialog(_ownerWindow));
 
+            // Setup a Task to listen for cancellation
+            // On Cancellation we need to Close the Dialog if it's open
+            // We also need to Stop the Task when the Dialog is finished.
+            var cts = new CancellationTokenSource();
+            
+            Task.Run(() =>
```

Comment:
Not sure about this. How about using  `using (token.Register(() => token.ThrowIfCancellationRequested())` ?

Reply:
I'm not sure about the Throwing, will that close the browser form? But I can certainly close the window using this and then throw... will adapt